core/src/components/datetime/datetime.tsx

```diff
@@ -1262,6 +1236,58 @@ export class Datetime implements ComponentInterface {
        */
       this.activeParts = [];
     }
+
+    /**
+     * Only animate if:
+     * 1. We're using grid style (wheel style pickers should just jump to new value)
+     * 2. The month and/or year actually changed (otherwise there's nothing to animate to)
+     * 3. The datetime is visible (prevents animation when in collapsed datetime-button, for example)
+     */
+    const didChangeMonth = month !== workingParts.month || year !== workingParts.year;
+    const elIsVisible = el.offsetParent !== null;
```

Comment:
The view does not update if the month/year picker is open on the grid style

Reply:
Fixed in https://github.com/ionic-team/ionic-framework/pull/27806/commits/7a3f92751372589c94d3cab63c915c64c4d60c77.